File: Leetcode/169-Majority-Element/MajorityElement.java
public class MajorityElement {
    
    public int majorityElement(int[] nums) {
        int number = 0;
        if (nums == null || nums.length == 0) {
            return number;
        }
        int count = 0;
        for (int i : nums) {
            if (i == number) {
                count++;
            } else {
                if (count == 0) {
                    number = i;
                } else {
                    count--;
                }
            }
        }
        return number;
    }
}

File: Leetcode/169-Majority-Element/MajorityElement.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    var res=0
    var count=0
    var num
    for (num in nums) {
        if (count==0) {
            res = nums[num]
            count++
        } else {
            if (nums[num]==res) {
                count++
            } else {
                count--
            }
        }
    }
    return res
};

